src/swap/SwapAmountInput.tsx

```diff
@@ -214,61 +218,51 @@ const styles = StyleSheet.create({
     paddingVertical: Spacing.Smallest8,
   },
   loaderContainer: {
-    position: 'absolute',
-    top: 0,
-    left: 0,
-    width: '100%',
-    height: '100%',
+    paddingVertical: Spacing.Small12,
   },
   loader: {
     height: '100%',
     width: '100%',
   },
-  fiatValueLoader: {
-    height: '100%',
-    width: '40%',
+  maxButtonWrapper: {
+    marginLeft: Spacing.Smallest8,
   },
   maxButton: {
     backgroundColor: Colors.white,
     borderWidth: 1,
     borderColor: Colors.gray2,
-    borderRadius: 4,
-    paddingVertical: 2,
-    paddingHorizontal: 4,
-    marginRight: Spacing.Smallest8,
+    borderRadius: Spacing.Tiny4,
+    paddingVertical: Spacing.Tiny4,
+    paddingHorizontal: 6,
   },
   maxText: {
-    ...fontStyles.small,
+    ...typeScale.labelXXSmall,
     color: Colors.gray5,
     fontSize: 10,
   },
-  tokenSelectButton: {
-    flexDirection: 'row',
-    alignItems: 'center',
-    backgroundColor: Colors.white,
-    borderWidth: 1,
-    borderColor: Colors.gray2,
-    borderRadius: 100,
-    paddingHorizontal: Spacing.Smallest8,
-    paddingVertical: 4,
-  },
   tokenName: {
-    ...fontStyles.small600,
+    ...typeScale.labelSemiBoldXSmall,
     paddingHorizontal: 4,
   },
-  tokenNamePlaceholder: {
-    ...fontStyles.small600,
+  tokenNetwork: {
+    ...typeScale.bodyXSmall,
+    color: Colors.gray4,
     paddingHorizontal: 4,
-    color: Colors.primary,
   },
-  tokenImage: {
-    width: 24,
-    height: 24,
-    borderRadius: 12,
+  tokenInfoText: {
+    paddingLeft: Spacing.Smallest8,
+  },
+  tokenNamePlaceholder: {
+    ...typeScale.labelMedium,
+    paddingHorizontal: 4,
+    color: Colors.gray3,
   },
   fiatValue: {
     ...typeScale.bodyXSmall,
+    paddingLeft: Spacing.Smallest8,
+    maxWidth: '40%',
```

Comment:
if we're adding a max width, we should also add a `numberOfLines={1}` so that the text is ellipsised when the limit is hit rather than spreading to new lines (e.g. if you type a long number like 123455678923624856 in the token amount - unlikely to happen but if the device font size is large then this number can become smaller to reach the limit)

Reply:
Aha, good catch.. I had tried to do this but was setting `numberOfLines={1}` at the wrong layer and it wasn't working as intended. Setting it on the `SwapAmountInput/FiatValue` component works well!